### PR TITLE
[fix] use non-inplace for insert observer pass

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34190 [fix] use non-inplace for insert observer pass**

Summary:
inplace modification of ClassType might affect other tests, so we want to do non-inplace modifications.
Actually the inplace argument will be removed soon.

Test Plan:
ci

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20451765](https://our.internmc.facebook.com/intern/diff/D20451765)

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -1023,7 +1023,7 @@ def forward(self, x):
         m = torch.jit.script(M())
         observer = torch.jit.script(default_observer())
         qconfig_dict = {'': script_qconfig(default_qconfig)}
-        torch._C._jit_pass_insert_observers(m._c, "forward", qconfig_dict, True)
+        m = wrap_cpp_module(torch._C._jit_pass_insert_observers(m._c, "forward", qconfig_dict, False))
```
Comment (raghuramank100) [inline test/test_jit.py]: Can you provide more context on why this change is needed? Only the test has been changed without any change in the code, so want to understand why.

Comment (jerryzh168) [reply test/test_jit.py]: this is make sure tests don't modify the original ClassType inplace so that it does not affect other tests in the system

**Review by raghuramank100 — APPROVED** [review]
